packages/adblocker-webextension/test/index.test.ts

```diff
@@ -125,3 +131,209 @@ describe('#fromWebRequestDetails', () => {
     });
   });
 });
+
+describe('html-filtering', () => {
+  context('#shouldApplyReplaceSelectors', () => {
+    function createWebRequestDetails({
+      responseType,
+      headers,
+    }: {
+      responseType: WebRequest.ResourceType;
+      headers: Record<string, string>;
+    }) {
+      const url = 'https://foo.com/';
+      const requestId = 'req-01';
+      const tabId = 1;
+      const details: OnHeadersReceivedDetailsType = {
+        requestId,
+        tabId,
+        url,
+        responseHeaders: Object.entries(headers).map(([name, value]) => ({
+          name,
+          value,
+        })),
+        type: responseType,
+      };
+
+      return details;
+    }
+
+    const sets: Array<Parameters<typeof createWebRequestDetails>[0] & { result: boolean }> = [
```

Comment:
I'm not sure if this meta programming brings much. Can we replace it with a separate `it` blocks, each having a proper description?

Reply:
Updated in [`9991a4a` (#14)](https://github.com/seia-soto/adblocker/pull/14/commits/9991a4a7979f149105991bd4b199be1250a86d41)